src/search.py

```diff
@@ -0,0 +1,148 @@
+from typing import Dict
+from typing import List
+
+import pandas as pd
+import yaml
+
+from src import logger
+
+
+def load_config(config_path: str) -> dict:
```

Comment:
If you wanted to make this more robust, it might be nice to add some checks here to ensure that the config contains the expected keys. Or you could move these checks inside the relevant functions - I guess if someone is only planning to do keyword searches, it doesn't matter if they haven't got "scope_statements" in the config?

Reply:
That's a good idea for a new feature - at the moment, the approach is stuck to one search pattern. Thank you!